File: program1/C/Colleen.c
#include <stdio.h>

/*
    First comment
*/

void describeFile() {
    char *program = "#include <stdio.h>%2$c%2$c/*%2$c    First comment%2$c*/%2$c%2$cvoid describeFile() {%2$c    char *program = %3$c%1$s%3$c;%2$c    printf(program, program, 10, 34);%2$c}%2$c%2$cint main() {%2$c    /*%2$c        Second Comment%2$c    */%2$c    describeFile();%2$c    return(0);%2$c}";
    printf(program, program, 10, 34);
}

int main() {
    /*
        Second Comment
    */
    describeFile();
    return(0);
}

File: program1/JS/Colleen.js
/*
    First comment
*/
function    decribeFile() {
    let carriageReturn = String.fromCharCode(10);
    let doubleQuotes = String.fromCharCode(34);
    let programContent = "/*%1    First comment%1*/%1function    decribeFile() {%1    let carriageReturn = String.fromCharCode(10);%1    let doubleQuotes = String.fromCharCode(34);%1    let programContent = %2%s%2;%1    let programReplace = %2%3%2;%1    let program = %3;%1    console.log(program, programContent);%1}%1%1function main(){%1    decribeFile();%1    /*%1        Second Comment%1    */%1}%1%1main();";
    let programReplace = "programContent.replace(/%1/g, carriageReturn).replace(/%2/g, doubleQuotes).replace(/%3/g, programReplace)";
    let program = programContent.replace(/%1/g, carriageReturn).replace(/%2/g, doubleQuotes).replace(/%3/g, programReplace);
    console.log(program, programContent);
}

function main(){
    decribeFile();
    /*
        Second Comment
    */
}

main();
